linux57-tkg/install.sh

```diff
@@ -132,74 +168,115 @@ if [ "$1" = "install" ]; then
 
   # ccache
   if [ "$_noccache" != "true" ]; then
-    if [ "$_distro" = "Ubuntu" ] && dpkg -l ccache > /dev/null; then
+
+    if [ "$_distro" = "Ubuntu" ] || [ "$_distro" = "Debian" ]; then
       export PATH="/usr/lib/ccache/bin/:$PATH"
-      export CCACHE_SLOPPINESS="file_macro,locale,time_macros"
-      export CCACHE_NOHASHDIR="true"
-      msg2 'ccache was found and will be used'
+    elif [ "$_distro" = "Fedora" ] || [ "$_distro" = "Suse" ]; then
+      export PATH="/usr/lib64/ccache/:$PATH" 
     fi
+
+    export CCACHE_SLOPPINESS="file_macro,locale,time_macros"
+    export CCACHE_NOHASHDIR="true"
+    msg2 'ccache was found and will be used'
+
   fi
 
-  _kernel_flavor="${_kernel_localversion}"
   if [ -z $_kernel_localversion ]; then
     _kernel_flavor="tkg-${_cpusched}"
+  else
+    _kernel_flavor="tkg-${_kernel_localversion}"
   fi
 
-  if [ "$_distro" = "Ubuntu" ]; then
+  if [ "$_distro" = "Ubuntu" ]  || [ "$_distro" = "Debian" ]; then
+
     if make -j ${_thread_num} deb-pkg LOCALVERSION=-${_kernel_flavor}; then
       msg2 "Building successfully finished!"
+
+      cd "$_where"
+
+      # Create DEBS folder if it doesn't exist
+      mkdir -p DEBS
+      
+      # Move rpm files to RPMS folder inside the linux-tkg folder
+      mv "$_where"/*.deb "$_where"/DEBS/
+
       read -p "Do you want to install the new Kernel ? y/[n]: " _install
-      if [[ $_install =~ [yY] ]] || [[ $_install =~ [yY] ]] || [ $_install = "yes" ] || [ $_install = "Yes" ]; then
+      if [[ $_install =~ [yY] ]] || [ $_install = "yes" ] || [ $_install = "Yes" ]; then
         cd "$_where"
         _kernelname=$_basekernel.$_kernel_subver-$_kernel_flavor
-        _headers_deb=linux-headers-${_kernelname}*.deb
-        _image_deb=linux-image-${_kernelname}_*.deb
+        _headers_deb="linux-headers-${_kernelname}*.deb"
+        _image_deb="linux-image-${_kernelname}_*.deb"
+        _kernel_devel_deb="linux-libc-dev_${_kernelname}*.rpm"
         
-        sudo dpkg -i $_headers_deb $_image_deb
+        cd DEBS
+        sudo dpkg -i $_headers_deb $_image_deb $_kernel_devel_deb
+      fi
+    fi
+
+  elif [[ "$_distro" = "Fedora" ||  "$_distro" = "Suse" ]]; then
+
+    # Replace dashes with underscores, it seems that it's being done by binrpm-pkg
+    # Se we can actually refer properly to the rpm files.
+    _kernel_flavor=${_kernel_flavor//-/_}
+
+    if make -j ${_thread_num} rpm-pkg EXTRAVERSION="_${_kernel_flavor}"; then
+      msg2 "Building successfully finished!"
 
-        # Add to the list of installed kernels, used for uninstall
-        if ! { [ -f installed-kernels ] && grep -Fxq "$_kernelname" installed-kernels; }; then
-          echo $_kernelname >> installed-kernels 
-        fi   
+      cd "$_where"
+
+      # Create RPMS folder if it doesn't exist
+      mkdir -p RPMS
+      
+      # Move rpm files to RPMS folder inside the linux-tkg folder
+      mv ~/rpmbuild/RPMS/x86_64/* "$_where"/RPMS/
```

Comment:
~/rpmbuild is the default %{_topdir} so it might also be used for other packages